### ISSUE TITLE
[Bug] Cannot disable shuffling with `prepare_data_loader`

### ISSUE THREAD

**Issue by bveeramani** [ray-project/ray#22406]
### Search before asking

- [X] I searched the [issues](https://github.com/ray-project/ray/issues) and found no similar issues.

### Ray Component

Ray Train

### What happened + What you expected to happen

I want to prepare my data loader for distributed inference, but I don't want to shuffle the samples (as is typical for model evaluation).

```
test_datatset = torchvision.datasets.ImageFolder("~/data/TinyImageNet/val")
test_data_loader = DataLoader(test_dataset, shuffle=False)
test_data_loader = prepare_data_loader(test_data_loader)  # This data loader will shuffle indices!
```

However, `prepare_data_loader` will always shuffle the indices unless I pass in a `SequentialSampler` to `prepare_data_loader`. The relevant code snippet from the source code is included below.

```
def with_sampler(loader):
        ...

        # Always evaluates to `True` unless a `SequentialSampler` is passed in!
        shuffle = not isinstance(loader.sampler, SequentialSampler)

        data_loader_args = {
            ...,
            "sampler": DistributedSampler(loader.dataset, shuffle=shuffle),
        }
        return DataLoader(**data_loader_args)

data_loader = with_sampler(data_loader)
```

My current work around is to manually create and pass in a `DistributedSampler`.

### Versions / Dependencies

Python: 3.8
OS: Ubuntu
Ray: 1.10

### Reproduction script

TK

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

**Comment by amogkam** [comment]
What version of torch is this?
